### PR TITLE
DHFPROD-2531: Add snackbar saved indicator

### PR DESCRIPTION
**Pull request by wooldridge**
Add snackbar saved indicator for UI changes that lack feedback:

- Ingestion step config
- Mapping step config
- Custom step config

### CODE REVIEW

File: web/src/main/ui/app/components/flows-new/edit-flow/custom/ui/custom-ui.component.ts [hunk 1]
```diff
@@ -11,16 +12,10 @@ export class CustomUiComponent {
...
-    if (event.key === 'Enter') {
-      this.onChange();
-    }
+  constructor(private snackBar: MatSnackBar) {
   }
 
   onChange() {
     this.updateCustom.emit(this.step);
-  }
+    this.snackBar.open("Change Saved", "", {panelClass: ['snackbar'], duration: 1200});  }
```
Comment (brucean52) [inline web/src/main/ui/app/components/flows-new/edit-flow/custom/ui/custom-ui.component.ts]: the snackbar open code should be in the edit-flow.component.ts where @briantang originally had it. This way the snackbar doesn't appear until we get the response from the server.

File: web/src/main/ui/app/components/flows-new/edit-flow/edit-flow.component.ts [hunk 2]
```diff
@@ -175,6 +192,9 @@ export class EditFlowComponent implements OnInit, OnDestroy {
     this.selectedStepId = this.stepsArray[index].id;
   }
   updateStep(step) {
+    if(step.stepDefinitionType !== this.stepType.MASTERING){
+      this.snackBar.open("Change Saved", "", {panelClass: ['snackbar'], duration: 1200});
+    }
     this.manageFlowsService.updateStep(this.flow.id, step.id, step).subscribe(resp => {
       this.stepsArray.forEach( step => {
```
Comment (brucean52) [inline web/src/main/ui/app/components/flows-new/edit-flow/edit-flow.component.ts]: snackbar code should be after the resp is received.

**Comment by wooldridge** [comment]
@ayuwono the mastering config is a bit different in that the user gets feedback upon saves through the dialog closure and the timestamp button changing. I need to discuss with PM.

**Comment by ayuwono** [comment]
@wooldridge, I think there are some conflicts to be resolved in order to merge this

**Comment by wooldridge** [comment]
Discussed with @sbayatpur. see: https://project.marklogic.com/jira/browse/DHFPROD-2531

 Added snackbar feedback for Mastering configuration. Now also step configuration has consistent saved feedback. Also updated snackbar labeling to "Step Saved".

**Comment by wooldridge** [comment]
Also moved snackbar generation to edit-flow-ui.component since it is a UI feature. It is now called from edit-flow.component via @ViewChild.

**Comment by wooldridge** [comment]
@ayuwono you bug you mention is separate from the feature in this PR. Please file a separate bug so we can merge this. Thanks for catching!

**Comment by ayuwono** [comment]
> @ayuwono you bug you mention is separate from the feature in this PR. Please file a separate bug so we can merge this. Thanks for catching!

sounds good, will log a separate issue

**Pull request merged**